---
float32[8] light_readings
float32 max_intensity